includes/admin/tools/class-edd-tools-recount-all-stats.php

```diff
@@ -72,73 +76,57 @@ public function get_data() {
 
 		$all_downloads = $this->get_stored_data( 'edd_temp_download_ids' );
 
-		$args = apply_filters( 'edd_recount_download_stats_args', array(
-			'post_parent__in' => $all_downloads,
-			'post_type'       => 'edd_log',
-			'posts_per_page'  => $this->per_step,
-			'post_status'     => 'complete',
-			'paged'           => $this->step,
-			'log_type'        => 'sale',
-			'fields'          => 'ids',
-		) );
-
-		$log_ids = $edd_logs->get_connected_logs( $args, 'sale' );
-
-		if ( $log_ids ) {
-			$log_ids     = implode( ',', $log_ids );
-			$payment_ids = $wpdb->get_col( "SELECT meta_value FROM $wpdb->postmeta WHERE meta_key='_edd_log_payment_id' AND post_id IN ($log_ids)" );
-			unset( $log_ids );
+		$deprecated_args = edd_apply_filters_deprecated( 'edd_recount_download_stats_args', array(
+			array(
+				'post_parent__in' => $all_downloads,
+				'post_type'       => 'edd_log',
+				'posts_per_page'  => $this->per_step,
+				'post_status'     => 'complete',
+				'paged'           => $this->step,
+				'log_type'        => 'sale',
+				'fields'          => 'ids',
+			)
+		), '3.0' );
+
+		$new_args = array(
+			'product_id__in' => $all_downloads,
+			'status__in'     => $accepted_statuses,
+			'number'         => $deprecated_args['posts_per_page'],
+			'offset'         => ( $deprecated_args['paged'] * $deprecated_args['posts_per_page'] ) - $deprecated_args['posts_per_page']
+		);
 
-			$payment_ids = implode( ',', $payment_ids );
-			$payments = $wpdb->get_results( "SELECT ID, post_status FROM $wpdb->posts WHERE ID IN (" . $payment_ids . ")" );
-			unset( $payment_ids );
+		$order_items = edd_get_order_items( $new_args );
 
-			foreach ( $payments as $payment ) {
+		if ( $order_items ) {
+			foreach ( $order_items as $order_item ) {
 
 				// Prevent payments that have all ready been retrieved from a previous sales log from counting again.
-				if ( in_array( $payment->ID, $processed_payments ) ) {
+				if ( in_array( $order_item->id, $processed_payments ) ) {
```

Comment:
Could this and the next `in_array` check include `true` as the third parameter?

Reply:
For this one, since it is coming from the database, there is the possibility that a database abstraction could possibly alter it from a string to an int or an int to a string. Since database dropins could modify the way serialization happens. I can't think of a certain situation where using `true` would benefit us further at this time, unless you have an example.

Leaving it without `true` for the other param means we'll be able to process either way.

Same goes for the accepted statuses. Since those can be filtered, I think it is safer to just leave them as 'loose' searches to account for possible data type changes. I can't imagine someone using numeric statuses, but it's not impossible that it could be an integer in the array above, but a string coming from the DB.